qiskit/providers/fake_provider/fake_job.py

```diff
@@ -19,13 +19,20 @@
 
 from qiskit.providers import JobV1
 from qiskit.providers.jobstatus import JobStatus
+from qiskit.utils.deprecation import deprecate_func
 
 
 class FakeJob(JobV1):
     """Fake simulator job"""
 
     _executor = futures.ThreadPoolExecutor()
 
+    @deprecate_func(
+        additional_msg="Use the `qiskit.providers.JobV1` class instead.",
```

Comment:
Just to be clear, we recommend that they use the base class to roll their own? If so, should we add the word "directly" in there somewhere, e.g. `"Use the qiskit.providers.JobV1 class directly instead."`